### ISSUE TITLE
Typescript example for use with <StoryblokComponent />

### ISSUE THREAD

**Issue by dasblitz** [storyblok/storyblok-react#257]
How can I use `<StoryblokComponent />`  with TypeScript and a type safe `blok` prop?

I have a page that renders products with a different datasource as storyblok, with some storyblok content in between those products.

It's a nextjs app so I have an _app.tsx where I define the mapping of the storyblok bloks to my react components:

```typescript
// in _app.tsx
const storyBlokComponents = {
  promoTile: PromoTile,
};

storyblokInit({
  accessToken: process.env.NEXT_PUBLIC_STORYBLOK_KEY,
  use: [apiPlugin],
  components: storyBlokComponents,
});

// etc...
```

My PromoTile is a component with it's props typed:

```tsx
// PromoTile.tsx
import { SbBlokData, storyblokEditable } from '@storyblok/react';

export type TPromoTile = {
  blok : {
    imageSrc: string;
  }
} & SbBlokData // is this even the correct way to type?

export const PromoTile ({ blok }: IPromoTile ) => {
  const {
    imageSrc,
   //..etc
  } = blok;
  return (
    <div {...storyblokEditable(blok)}>
      <img src={imageSrc}>
    </div>
  )
}
```

In the component that renders the products I use the `<StoryblokComponent />` but it will never show an error if I pass in something to blok thats not of type `TPromoTile` but is of type `SbBlokData` :

```tsx
// ProductList.tsx
type ProductListProps = {
  products: [],
  promoTiles: TPromoTile[]
}

export const ProductList: ({products, promoTiles} : ProductListProps) => {
  return products.map((product, index) => (
    <React.Fragment key={product.id}>
      <ProductTile product={product} />
      
      {/* I can add any object as the blok prop that adheres to SbBlokData type */}
      <StoryblokComponent blok={promoTiles[index]} key={promoTile._uid} />
     </React.Fragment>
  ))
}
```
So long story short, should I be using `<StoryblokComponent />` if I want type safety and if so, how can I make it typesafe? 
Or should I just use `<PromoTile blok={promoTiles[index]} key={promoTile._uid} />` instead?

**Comment by fgiuliani** [comment]
Hi @dasblitz!

If your `ProductList` will always be a list of `PromoTile` components, you can use that React Component directly. `StoryblokComponent` is used for dynamic rendering, when the blocks/components that come from Storyblok API can have different types
